dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/generator/ForwardGenerator.java

```diff
@@ -101,7 +103,27 @@ public void endGeneration(int section, JavaCodeWriter writer)  throws JspCoreExc
                     persistentData.put("methodNesting", new Integer(0));
                 }
                 int methodNesting =  ((Integer)persistentData.get("methodNesting")).intValue();
-                writer.println((methodNesting > 0) ? "return true;" : "return;");
+                if (methodNesting > 0) {
+                    // pull in parentTagName in order to clean up CDI
```

Comment:
as I recalled, the parent tag (i.e the dispatcher tag) does have these destroy() but it was skipped.  I dont think we should aggressively doDestroy for parent, from the child tag.

Reply:
Unfortunately the way the code generator is currently structured, you have to add the PreDestroy where otherwise it was generating a bare return.  Generating it anywhere else up the call chain would make the determination logic more complex.